src/Database/Relations/BelongsToMany.php

```diff
@@ -188,7 +188,7 @@ public function detach($ids = null, $touch = true)
         /*
          * See Illuminate\Database\Eloquent\Relations\Concerns\InteractsWithPivotTable
```

Comment:
Can you update all the relation before* events to not be halting events and instead require exceptions to be thrown to halt them?

Reply:
might just be the one beforeAttach one above

Reply:
This comment is meant for #537 btw